src/fd.c

```diff
@@ -0,0 +1,5 @@
+#include <fcntl.h>
+
+int ro_fd (const char *path) {
+    return open (path, O_RDONLY);
```

Comment:
`Posix.open` is the vala binding of the same method, any reason why you're exposing it from C?